libs/domain/picking/src/components/picking/picking.component.ts

```diff
@@ -1,53 +1,142 @@
-import { asyncState, i18n, valueType } from '@spryker-oryx/utilities';
+import {
+  asyncState,
+  i18n,
+  subscribe,
+  valueType,
+} from '@spryker-oryx/utilities';
 import { html, LitElement, TemplateResult } from 'lit';
+import { when } from 'lit-html/directives/when.js';
+import { state } from 'lit/decorators.js';
 import { repeat } from 'lit/directives/repeat.js';
-import { map } from 'rxjs';
+import { BehaviorSubject, take, tap } from 'rxjs';
 import { PickingListMixin } from '../../mixins';
-import { ItemsFilters } from '../../models';
-
+import {
+  ItemsFilters,
+  PartialPicking,
+  PickingListItem,
+  PickingTab,
+  ProductItemPickedEvent,
+} from '../../models';
+import { styles } from './picking.styles';
 export class PickingComponent extends PickingListMixin(LitElement) {
-  protected tabs$ = this.pickingList$.pipe(
-    map((list) => [
+  static styles = styles;
+
+  protected isConfirmPickingDialogOpen$ = new BehaviorSubject(false);
+  @asyncState()
+  protected isConfirmPickingDialogOpen = valueType(
+    this.isConfirmPickingDialogOpen$
+  );
+
+  @state()
+  protected partialPicking: PartialPicking | undefined;
```

Comment:
`protected partialPicking: PartialPicking | null = null;`

You can use this one as a trigger for confirmation modal. It will contain partial picking data or null. When it has data -> show modal. And you can clean up `isConfirmPickingDialogOpen`